### PR TITLE
Response Cache Optimizations

### PR DESCRIPTION
**Pull request by JoeHegarty**
This resolves an issue discovered by @ceckhardt's team which causes thread contention in the Response Cache system. 
This is caused by locking internally in the JDK when creating a new message digest. It appears this is a known issue and has been worked around in other projects such as Guava and the recommendation is to clone the digest.

### CODE REVIEW

**Comment by jpennell** [comment]
I'll pull this and re-run our load tests!

File: actors/runtime/src/main/java/cloud/orbit/actors/Stage.java [hunk 1]
```diff
@@ -715,7 +715,7 @@ public void setDeactivationTimeout(long deactivationTimeoutMs)
                 .findFirst()
                 .orElseGet(() ->
                 {
-                    final ResponseCaching responseCaching = new ResponseCaching();
+                    final DefaultResponseCachingExtension responseCaching = new DefaultResponseCachingExtension();
```
Comment (ShawnTuatara) [inline actors/runtime/src/main/java/cloud/orbit/actors/Stage.java]: Syntax sugar... make the responseCaching of type ResponseCachingExtension (coding to interfaces)?

Comment (johnou) [inline actors/runtime/src/main/java/cloud/orbit/actors/Stage.java]: +1

File: commons/src/main/java/cloud/orbit/concurrent/MessageDigestFactory.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Created by joeh on 2017-06-27.
+ */
+public class MessageDigestFactory
+{
+    private final String algorithm;
+    private final MessageDigest digestPrototype;
+    private final Boolean supportsClone;
```
Comment (johnou) [inline commons/src/main/java/cloud/orbit/concurrent/MessageDigestFactory.java]: use boolean primitive?

**Pull request merged**